src/Tempest/Console/src/Console.php

```diff
@@ -56,4 +56,12 @@ public function success(string $line): self;
     public function when(mixed $expression, callable $callback): self;
 
     public function withLabel(string $label): self;
+
+    public function supportsTty(): bool;
```

Comment:
Technically, the right place for TTY stuff would be in the `Terminal` class, not on the `Console`. Think of the `Terminal` is the car's engine, while the `Console` adds the features to interact with it.